test/integration/boundingbox_camera_plugin.cc

```diff
@@ -0,0 +1,474 @@
+  public testing::WithParamInterface<const char *>
+{
+  // Create a BoundingBox Camera sensor from a SDF and gets a boxes message
+  public: void BoxesWithBuiltinSDF(const std::string &_renderEngine);
+
+  public: void Boxes3DWithBuiltinSDF(const std::string &renderEngine);
+};
+
+/// \brief mutex for thread safety
+std::mutex g_mutex;
+
+/// \brief bounding boxes from the camera
+std::vector<BoundingBox> g_boxes;
+
+/// \brief counter of received boundingboxes msg
+int g_counter = 0;
+
+/// \brief callback to receive 2d boxes from the camera
+void OnNewBoundingBoxes(const msgs::AnnotatedAxisAligned2DBox_V &boxes)
+{
+  g_mutex.lock();
+  g_boxes.clear();
+
+  int size = boxes.annotated_box_size();
+  for (int i = 0; i < size; i++)
+  {
+    BoundingBox box(BoundingBoxType::VisibleBox2D);
+    auto annotated_box = boxes.annotated_box(i);
+    auto axisAlignedBox = annotated_box.box();
+    auto min_corner = axisAlignedBox.min_corner();
+    auto max_corner = axisAlignedBox.max_corner();
+
+    box.size.X(max_corner.x() - min_corner.x());
+    box.size.Y(max_corner.y() - min_corner.y());
+    box.size.Z(0);
+
+    box.center.X(min_corner.x() + box.size.X() / 2);
+    box.center.Y(min_corner.y() + box.size.Y() / 2);
+    box.center.Z(0);
+
+    box.oreintation.Set(0, 0, 0, 0);
+    box.label = annotated_box.label();
+    g_boxes.push_back(box);
+  }
+  g_counter++;
+  g_mutex.unlock();
+}
+
+/// \brief callback to receive 3D boxes from the camera
+void OnNew3DBoundingBoxes(const msgs::AnnotatedOriented3DBox_V &boxes)
+{
+  g_mutex.lock();
+  g_boxes.clear();
+
+  int size = boxes.annotated_box_size();
+  for (int i = 0; i < size; i++)
+  {
+    BoundingBox box(BoundingBoxType::Box3D);
+    auto annotated_box = boxes.annotated_box(i);
+    auto orientedBox = annotated_box.box();
+    // center
+    box.center.Set(orientedBox.center().x(), orientedBox.center().y(),
+      orientedBox.center().z());
+    // orientation
+    auto orientation = orientedBox.orientation();
+    box.oreintation.Set(orientation.w(), orientation.x(),
+      orientation.y(), orientation.z());
+    // size
+    auto boxSize = orientedBox.boxsize();
+    box.size.Set(boxSize.x(), boxSize.y(), boxSize.z());
+
+    box.label = annotated_box.label();
+    g_boxes.push_back(box);
+  }
+  g_counter++;
+  g_mutex.unlock();
+}
+
+/// \brief wait till you read the published frame
+void WaitForNewFrame()
+{
+  auto waitTime = std::chrono::duration_cast< std::chrono::milliseconds >(
+  std::chrono::duration< double >(0.001));
+  int counter = 0;
+  // wait for the counter to increase
+  for (int sleep = 0; sleep < 300 && counter == 0; ++sleep)
+  {
+    g_mutex.lock();
+    counter = g_counter;
+    g_mutex.unlock();
+    std::this_thread::sleep_for(waitTime);
+  }
+  EXPECT_GT(counter, 0);
+}
+
+/// \brief Build a scene with 3 boxes 2 overlapping boxes which 1
+/// is behind the other, and the 3rd box is invisible behind them
+void BuildScene(rendering::ScenePtr scene)
```

Comment:
```suggestion
void BuildScene2d(rendering::ScenePtr _scene)
```
Since the other method is called `BuildScene3d`, I am assuming that this is building scenes for `2d`?